test/pkg/test/framework.go

```diff
@@ -138,17 +138,30 @@ func (env *Framework) ExpectToBeClean(ctx context.Context) {
 				}
 			}
 		}
-	}).Should(Succeed())
-
-	//Currently, We have a know issue that the controller will not clear the TargetGroup when it cleans the k8s service
-	//Also, TargetGroup don't have an appropriate tag to identify whether this tg is created by current EKS Cluster or not.
-	//So, we temporarily don't check whether the TargetGroup have been cleaned.
 
-	//retrievedTargetGroups, _ := env.LatticeClient.ListTargetGroupsAsList(ctx, &vpclattice.ListTargetGroupsInput{})
-	//for _, tg := range retrievedTargetGroups {
-	//	Logger(ctx).Infof("Found TargetGroup: %v, checking it whether it's created by current EKS Cluster", tg)
-	//	g.Expect(*tg.Name).To(Not(ContainElements(BeKeyOf(env.TestCasesCreatedServiceNames))))
-	//}
+		retrievedTargetGroups, _ := env.LatticeClient.ListTargetGroupsAsList(ctx, &vpclattice.ListTargetGroupsInput{})
+		for _, tg := range retrievedTargetGroups {
+			Logger(ctx).Infof("Found TargetGroup: %v, checking it whether it's created by current EKS Cluster", tg)
+			if currentClusterVpcId != *tg.VpcIdentifier {
+				//This tg is not created by current EKS Cluster, skip it
+				continue
+			}
+			retrievedTags, err := env.LatticeClient.ListTagsForResourceWithContext(ctx, &vpclattice.ListTagsForResourceInput{
+				ResourceArn: tg.Arn,
+			})
+			if err == nil {
+				Logger(ctx).Infof("Found Tags for tg %v tags: %v", *tg.Name, retrievedTags)
+				tagValue, ok := retrievedTags.Tags[lattice.K8SParentRefTypeKey]
+				if ok && *tagValue == lattice.K8SServiceExportType {
+					//This tg is created by k8s controller, by a ServiceExport,
+					//ServiceExport still have a known targetGroup leak issue,
+					//so we temporarily skip to verify whether ServiceExport created TargetGroup is deleted or not
+					continue
+				}
+				Expect(*tg.Name).To(Not(ContainElements(BeKeyOf(env.TestCasesCreatedServiceNames))))
+			}
```

Comment:
New test framework works for the new controller code, it could correct skip other cluster's tg and serviceExport created tg:


```
   logger.go:130: 2023-06-15T19:49:05.341-0700	INFO	test/framework.go:144	Found TargetGroup: {
          Arn: "arn:aws:vpc-lattice:us-west-2:348739612106:targetgroup/tg-02b3e0afe03a6cf58",
          CreatedAt: 2023-06-16 02:40:29.099 +0000 UTC,
          Id: "tg-02b3e0afe03a6cf58",
          IpAddressType: "IPV4",
          LastUpdatedAt: 2023-06-16 02:40:29.099 +0000 UTC,
          Name: "k8s-glaze-11-qzxpkn4pvo-default",
          Port: 80,
          Protocol: "HTTP",
          ServiceArns: [],
          Status: "ACTIVE",
          Type: "IP",
          VpcIdentifier: "vpc-0c9c66ebcf3b99ae2"
        }, checking it whether it's created by current EKS Cluster
    logger.go:130: 2023-06-15T19:49:05.459-0700	INFO	test/framework.go:153	Found Tags for tg k8s-glaze-11-qzxpkn4pvo-default tags: {
          Tags: {
            K8SServiceName: "glaze-11-qzxpkn4pvo",
            K8SParentRefTypeKey: "K8SServiceExportType",
            K8SServiceNamespace: "default"
          }
        }
    logger.go:130: 2023-06-15T19:49:05.459-0700	INFO	test/framework.go:144	Found TargetGroup: {
          Arn: "arn:aws:vpc-lattice:us-west-2:348739612106:targetgroup/tg-0f8f06ea953ad8903",
          CreatedAt: 2023-06-09 16:26:47.128 +0000 UTC,
          Id: "tg-0f8f06ea953ad8903",
          IpAddressType: "IPV4",
          LastUpdatedAt: 2023-06-09 16:26:47.128 +0000 UTC,
          Name: "k8s-parking-ver1-default",
          Port: 8090,
          Protocol: "HTTP",
          ServiceArns: [],
          Status: "ACTIVE",
          Type: "IP",
          VpcIdentifier: "vpc-0b46728134fe4943f"
        }, checking it whether it's created by current EKS Cluster
• [253.781 seconds]
------------------------------

Ran 3 of 3 Specs in 772.235 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestIntegration (772.66s)
PASS
ok  	github.com/aws/aws-application-networking-k8s/test/suites/integration	773.397s
```
<img width="1631" alt="image" src="https://github.com/aws/aws-application-networking-k8s/assets/32318664/a6eaf009-836f-4616-a09a-07b5776dca63">
<img width="1662" alt="image" src="https://github.com/aws/aws-application-networking-k8s/assets/32318664/2ba84d60-3347-4760-9ba3-d168187a0baf">